File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/jdk8/Colors.java
package com.xebia.functional.xef.java.auto.jdk8;

import com.xebia.functional.xef.conversation.PlatformConversation;
import com.xebia.functional.xef.conversation.llm.openai.OpenAI;
import com.xebia.functional.xef.prompt.Prompt;

import java.util.List;
import java.util.concurrent.ExecutionException;

public class Colors {

    public List<String> colors;

    public static void main(String[] args) throws ExecutionException, InterruptedException {
        PlatformConversation scope = OpenAI.conversation();
        scope.prompt(OpenAI.FromEnvironment.DEFAULT_SERIALIZATION, new Prompt("a selection of 10 beautiful colors that go well together"), Colors.class)
                  .thenAccept(colors -> System.out.println("Colors:\n" + colors.colors))
                  .get();

    }
}

File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/conversation/Colors.kt
package com.xebia.functional.xef.conversation

import com.xebia.functional.xef.conversation.llm.openai.OpenAI
import com.xebia.functional.xef.conversation.llm.openai.prompt
import kotlinx.serialization.Serializable

@Serializable data class Colors(val colors: List<String>)

suspend fun main() {
  OpenAI.conversation {
    val colors: Colors = prompt("a selection of 10 beautiful colors that go well together")
    println(colors)
  }
}
